.scratch/chatgpt-airgap-unlock/research/13-test-production-evidence.py

```
@@ -0,0 +1,797 @@
+
+            self.assertEqual(production_evidence.EVIDENCE_FAILURE, result)
+            verdict = json.loads(
+                (evidence / "verdict.json").read_text(encoding="utf-8")
+            )
+            self.assertFalse(verdict["cleanup_state_valid"])
+            self.assertIn("owned-state-invalid", verdict["errors"])
+
+    def test_evidence_paths_inside_staging_tree_are_rejected(self) -> None:
+        with tempfile.TemporaryDirectory() as temporary:
+            stage, manifest, state = self.make_run(temporary)
+            with self.assertRaises(production_evidence.EvidenceError):
```

Ruff (0.15.21)

[warning] Use `pytest.raises` instead of unittest-style `assertRaises`

Replace `assertRaises` with `pytest.raises`

(PT027)